### PR TITLE
Base type and implementation of unionSplit

### LINKED ISSUE #4: Feature request: Split array given another array into union sections.
Given the types array, it would be nice if there were a way to compare with another array and get the following

- Elements from the first array that doesn't match the comparator.
- Elements from the second array that doesn't match the comparator.
- Elements from the first array that matches the comparator.

Idea usage

```typescript
type Users = {
  id: string;
  name: string;
}

const users: User[] = [...];
const userIdsToDelete: string[];

const uninionSplits = users.unionSplit(userIdsToDelete, ((a: User, b: string) => a.id === b))

const {  
  leftOnly,
  rightOnly,
  union
} = uninionSplits;
```

### PR DESCRIPTION
**Pull request by objmj-itminds**
Implementation of issue #4.
Resolves #4

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -189,7 +189,7 @@ interface Array<T> {
    */
   findAndReplace(
     predicate: (value: T, index: number, obj: T[]) => boolean,
-    replaceVal: T,
+    replaceVal: T | ((x: T) => Partial<T>),
```
Comment (phillip-itminds) [inline src/types.ts:192]: You should probably create a test for using the function instead of T as replace value as well.

**Review by phillip-itminds — APPROVED** [review]
Other than my single comment, I think it looks good!

File: src/types.ts (hunk 1 above)
Comment (objmj-itminds) [reply src/types.ts:192]: It wasn't meant to be part of this PR. I'll remove it.

File: tests/index.test.ts [hunk 2]
```diff
@@ -295,3 +295,16 @@ describe("findAndReplace", () => {
...
+describe("unionSplit", () => {
+  const fruitsToCheck = ["banana", "strawberry", "lemon"];
+
+  it("Who's fruit is it?", () => {
+    const result = myArr.unionSplit(
+      fruitsToCheck,
+      (a, b) => a.favoriteFruit === b
+    );
+
+    console.log("unionSplit", result);
```
Comment (objmj-itminds) [inline tests/index.test.ts:308]: Oops. I was meant to use Expect here after logging the initial result to eye test it.

**Pull request merged**